sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/ToNumberParser.scala

```diff
@@ -49,33 +49,58 @@ object ToNumberParser {
   final val WRAPPING_ANGLE_BRACKETS_TO_NEGATIVE_NUMBER_END = 'R'
 
   // This class represents one or more characters that we expect to be present in the input string
-  // based on the format string.
-  abstract class InputToken()
+  // based on the format string. The toString method returns a representation of each token suitable
+  // for use in error messages.
+  abstract class InputToken() {
+    def toString: String
```

Comment:
we don't need this, `toString` is built-in API for all java classes.

Reply:
that said, we can override `toString` in subclasses already.

Reply:
Good point, done.